engines/wintermute/detection.cpp

```diff
@@ -210,6 +214,1425 @@ class WintermuteMetaEngine : public AdvancedMetaEngine {
+			act->addDefaultInputMapping("HASH"); // original keyboard
+			gameKeyMap->addAction(act);
+
+			act = new Action("WTF", _("???"));
+			act->setKeyEvent(KEYCODE_HOME);
+			act->addDefaultInputMapping("HOME"); // original keyboard
+//TODO: extra joy control, e.g. "JOY_R+JOY_B"
+			gameKeyMap->addAction(act);
+		} else if (gameId == "alphapolaris") {
+			act = new Action("HINT", _("Show hints"));
+			act->setKeyEvent(KeyState(KEYCODE_SPACE, ASCII_SPACE));
+			act->addDefaultInputMapping("SPACE"); // original keyboard
+			act->addDefaultInputMapping("MOUSE_MIDDLE"); // extra mouse
+			act->addDefaultInputMapping("JOY_Y"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("INV", _("Show inventory"));
+			act->setKeyEvent(KeyState(KEYCODE_i, 'i'));
+			act->addDefaultInputMapping("i"); // original keyboard
+			act->addDefaultInputMapping("JOY_LEFT"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("HELP", _("Show help"));
+			act->setKeyEvent(KEYCODE_F1);
+			act->addDefaultInputMapping("F1"); // original keyboard
+			act->addDefaultInputMapping("JOY_RIGHT"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("SCRLUP", _("Scroll up"));
+			act->setMouseWheelUpEvent();
+			act->addDefaultInputMapping("MOUSE_WHEEL_UP"); // original mouse
+			act->addDefaultInputMapping("PAGEUP"); // extra keyboard
+			act->addDefaultInputMapping("JOY_UP"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("SCRLDN", _("Scroll down"));
+			act->setMouseWheelDownEvent();
+			act->addDefaultInputMapping("MOUSE_WHEEL_DOWN"); // original mouse
+			act->addDefaultInputMapping("PAGEDOWN"); // extra keyboard
+			act->addDefaultInputMapping("JOY_DOWN"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("GEOM", _("Show scene geometry"));
+			act->setKeyEvent(KeyState(KEYCODE_F2, 0, KBD_CTRL));
+			act->addDefaultInputMapping("C+F2"); // original keyboard
+			gameKeyMap->addAction(act);
+
+			act = new Action("SHADOW", _("Change shadow type"));
+			act->setKeyEvent(KeyState(KEYCODE_F3, 0, KBD_CTRL));
+			act->addDefaultInputMapping("C+F3"); // original keyboard
+			gameKeyMap->addAction(act);
+
+			act = new Action("SCRS", _("Save screenshot"));
+			act->setKeyEvent(KeyState(KEYCODE_F5, 0, KBD_CTRL));
+			act->addDefaultInputMapping("C+F5"); // original keyboard
+			gameKeyMap->addAction(act);
+
+			act = new Action("VOLMAX", _("Volume max"));
+			act->setKeyEvent(KeyState(KEYCODE_F6, 0, KBD_CTRL));
+			act->addDefaultInputMapping("C+F6"); // original keyboard
+			gameKeyMap->addAction(act);
+
+			act = new Action("DBGCLI", _("Show debug parser"));
+			act->setKeyEvent(KeyState(KEYCODE_F7, 0, KBD_CTRL));
+			act->addDefaultInputMapping("C+F7"); // original keyboard
+			act->addDefaultInputMapping("C+F10"); // original keyboard
+//TODO: extra joy control, e.g. "JOY_R+JOY_B"
+			gameKeyMap->addAction(act);
+
+			act = new Action("DBG", _("Debug print"));
```

Comment:
Renamed to "DBGTXT", thanks for pointing this.
Should we introduce some conventions to avoid things like this in the future? Like, "non-standard actions IDs should always be 6+ characters long"?

Reply:
Well the action ids are namespaced by the keymap names. There was no real conflict here. The only issue is if a backend remaps a standard action that happens to have the same name as game action. I'm not sure it's worth the effort of enforcing a naming convention.